--- a/files-ucode/usr/share/ucode/wifi/supplicant.uc
+++ b/files-ucode/usr/share/ucode/wifi/supplicant.uc
@@ -229,7 +229,7 @@ function setup_sta(data, config) {
 		 ]);
 	network_append_vars(config, [
 		'rsn_overriding', 'scan_ssid', 'noscan', 'disabled', 'multi_ap_profile', 'multi_ap_backhaul_sta',
-		'ocv', 'beacon_prot', 'key_mgmt', 'sae_pwe', 'psk', 'sae_password', 'pairwise', 'group', 'bssid',
+		'ocv', 'beacon_prot', 'key_mgmt', 'sae_pwe', 'sae_groups', 'psk', 'sae_password', 'pairwise', 'group', 'bssid',
 		'proto', 'mesh_fwding', 'mesh_rssi_threshold', 'frequency', 'fixed_freq',
 		'disable_ht', 'disable_ht40', 'disable_vht', 'vht', 'max_oper_chwidth',
 		'ht40', 'beacon_int', 'ieee80211w', 'rates', 'mesh_basic_rates', 'mcast_rate',
--- a/files-ucode/usr/share/schema/wireless.wifi-iface.json
+++ b/files-ucode/usr/share/schema/wireless.wifi-iface.json
@@ -1005,6 +1005,13 @@
 			"type": "number",
 			"enum": [ 0, 1, 2 ]
 		},
+		"sae_groups": {
+			"description": "SAE group list",
+			"type": "array",
+			"items": {
+				"type": "string"
+			}
+		},
 		"sae_require_mfp": {
 			"description": "Require MFP for all associations using SAE",
 			"type": "boolean"
